cylc/flow/task_trigger.py

```diff
@@ -147,7 +158,24 @@ def __str__(self):
         else:
             return '%s:%s' % (self.task_name, self.output)
 
-    __repr__ = __str__
+    def __repr__(self) -> str:
+        return f"<{type(self).__name__} {self}>"
+
+    def __hash__(self) -> int:
+        return hash((
+            self.task_name,
+            self.cycle_point_offset,
+            self.output,
+            self.offset_is_irregular,
+            self.offset_is_from_icp,
+            self.offset_is_absolute,
+            self.initial_point,
+        ))
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, TaskTrigger):
+            return NotImplemented
+        return hash(self) == hash(other)
```

Comment:
It's a dumb test to have to write, but it would make codecov happy...

Reply:
I've added a unit test as opposed to doctest